java/hiro-client/src/main/java/co/arago/hiro/client/rest/DefaultWebSocketClient.java

```diff
@@ -0,0 +1,238 @@
+package co.arago.hiro.client.rest;
+
+import co.arago.hiro.client.api.TokenProvider;
+import co.arago.hiro.client.api.WebSocketClient;
+import co.arago.hiro.client.util.Helper;
+import co.arago.hiro.client.util.HiroCollections;
+import co.arago.hiro.client.util.HiroException;
+import co.arago.hiro.client.util.Listener;
+import co.arago.hiro.client.util.Throwables;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import net.minidev.json.JSONValue;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.ws.WebSocket;
+import org.asynchttpclient.ws.WebSocketListener;
+import org.asynchttpclient.ws.WebSocketUpgradeHandler;
+
+public final class DefaultWebSocketClient implements WebSocketClient {
+
+  private static final int MAX_RETRIES = 5;
+  public static final String DEFAULT_API_VERSION = "6.1";
```

Comment:
api path must be adjusted. If I'm not mistaken it used to be `/_g` without any version.

Reply:
implemented with websocket type